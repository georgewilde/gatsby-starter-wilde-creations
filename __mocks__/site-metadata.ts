import { SiteMetadata } from '@context/site-metadata';

export const SiteMetadataMock: SiteMetadata = {
  companyName: 'Acme Co.',
  description: '',
  author: '',
};
