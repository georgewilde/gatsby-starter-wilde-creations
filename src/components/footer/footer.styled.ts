import styled from 'styled-components';

export const StyledFooter = styled.footer`
  display: flex;
  justify-content: space-between;
`;
