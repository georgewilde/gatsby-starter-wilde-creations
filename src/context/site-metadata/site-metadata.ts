import { createContext } from 'react';

export interface SiteMetadata {
  companyName: string;
  description: string;
  author: string;
}

const defaultState: SiteMetadata = {
  companyName: '',
  description: '',
  author: '',
};

export const SiteMetadataContext = createContext(defaultState);
