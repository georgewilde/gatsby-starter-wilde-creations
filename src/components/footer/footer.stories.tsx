import React from 'react';

import { Footer } from './footer';
import { SiteMetadataContext } from '@context/site-metadata';
import { SiteMetadataMock } from '@mocks';

export default {
  title: 'Simple Components/Footer',
  component: Footer,
};

export const Default = () => (
  <SiteMetadataContext.Provider value={SiteMetadataMock}>
    <Footer />
  </SiteMetadataContext.Provider>
);
