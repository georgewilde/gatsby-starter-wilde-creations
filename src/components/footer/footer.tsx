import React, { FC } from 'react';

import { StyledFooter } from './footer.styled';
import { SiteMetadataContext } from '@context/site-metadata';

export const Footer: FC = () => (
  <SiteMetadataContext.Consumer>
    {(siteMetadata): JSX.Element => (
      <StyledFooter>
        <div data-testid="copyright">
          Â© {new Date().getFullYear()}, {siteMetadata.companyName}
        </div>
        <div>
          Website created by{' '}
          <a
            href="https://www.wildecreations.co.uk"
            target="_blank"
            rel="noopener noreferrer"
            data-testid="createdBy"
          >
            Your Company Ltd
          </a>
        </div>
      </StyledFooter>
    )}
  </SiteMetadataContext.Consumer>
);
