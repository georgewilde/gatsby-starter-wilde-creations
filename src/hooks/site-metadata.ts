import { graphql, useStaticQuery } from 'gatsby';

import { SiteMetadata } from '@context/site-metadata';

interface SiteMetadataResult {
  site: {
    siteMetadata: SiteMetadata;
  };
}

export const useSiteMetadata = (): SiteMetadata => {
  const { site } = useStaticQuery<SiteMetadataResult>(
    graphql`
      query SiteMetaData {
        site {
          siteMetadata {
            companyName
            description
            author
          }
        }
      }
    `
  );

  return site.siteMetadata;
};
