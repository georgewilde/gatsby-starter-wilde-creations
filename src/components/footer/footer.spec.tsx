import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';

import { Footer } from './footer';
import { SiteMetadata, SiteMetadataContext } from '@context/site-metadata';

describe(`Footer component`, () => {
  let siteMetadata: SiteMetadata;

  beforeEach(() => {
    siteMetadata = {
      companyName: 'Acme Co.',
      description: '',
      author: '',
    };
  });

  it(`should display the current year as part of the copyright notice`, () => {
    const expected = new Date().getFullYear().toString();

    const { getByTestId } = render(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Footer />
      </SiteMetadataContext.Provider>
    );

    expect(getByTestId('copyright')).toHaveTextContent(expected);
  });

  it(`should display the site title as part of the copyright notice`, () => {
    const expected = 'Acme Co.';

    const { getByTestId } = render(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Footer />
      </SiteMetadataContext.Provider>
    );

    expect(getByTestId('copyright')).toHaveTextContent(expected);
  });

  it(`should open a new tab when clicking on the "website created by" link`, () => {
    const { getByTestId } = render(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Footer />
      </SiteMetadataContext.Provider>
    );

    expect(getByTestId('createdBy')).toHaveAttribute(
      'href',
      'https://www.wildecreations.co.uk'
    );
    expect(getByTestId('createdBy')).toHaveAttribute('target', '_blank');
  });

  it(`should render correctly`, () => {
    const tree = create(
      <SiteMetadataContext.Provider value={siteMetadata}>
        <Footer />
      </SiteMetadataContext.Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
