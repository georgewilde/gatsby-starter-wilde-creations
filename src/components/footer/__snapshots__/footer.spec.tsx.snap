// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<footer
  className="c0"
>
  <div
    data-testid="copyright"
  >
    Â© 
    2020
    , 
    Acme Co.
  </div>
  <div>
    Website created by
     
    <a
      data-testid="createdBy"
      href="https://www.wildecreations.co.uk"
      rel="noopener noreferrer"
      target="_blank"
    >
      Wilde Creations Ltd
    </a>
  </div>
</footer>
`;
